packages/postgress/src/MessagePickupRepositoryPg.ts

```diff
@@ -0,0 +1,616 @@
+          // If it doesn't exist, create the table.
+          await dbClient.query(createTableLive)
+          this.logger?.info(`[buildPgDatabase] PostgresDbService Table "${tableNameLive}" created.`)
+        } else {
+          // If the table exists, clean it (truncate or delete, depending on your requirements).
+          await dbClient.query(`TRUNCATE TABLE ${tableNameLive}`)
+          this.logger?.info(`[buildPgDatabase] PostgresDbService Table "${tableNameLive}" cleared.`)
+        }
+      } finally {
+        await dbClient.end()
+      }
+    } catch (error) {
+      this.logger?.error(`[buildPgDatabase] PostgresDbService Error creating database: ${error}`)
+    } finally {
+      await client.end()
+    }
+  }
+
+  /**
+   * This function checks that messages from the connectionId, which were left in the 'sending'
+   * state after a liveSessionRemove event, are updated to the 'pending' state for subsequent sending
+   * @param connectionID
+   */
+
+  private async checkQueueMessages(connectionID: string): Promise<void> {
+    try {
+      this.logger?.debug(`[checkQueueMessages] Init verify messages state 'sending'`)
+      const messagesToSend = await this.messagesCollection?.query(
+        'SELECT * FROM queuedmessages WHERE state = $1 and connectionid = $2',
+        ['sending', connectionID],
+      )
+      if (messagesToSend && messagesToSend.rows.length > 0) {
+        for (const message of messagesToSend.rows) {
+          // Update the message state to 'pending'
+          await this.messagesCollection?.query('UPDATE queuedmessages SET state = $1 WHERE id = $2', [
+            'pending',
+            message.id,
+          ])
+        }
+
+        this.logger?.debug(`[checkQueueMessages] ${messagesToSend.rows.length} messages updated to 'pending'.`)
+      } else {
+        this.logger?.debug('[checkQueueMessages] No messages in "sending" state.')
+      }
+    } catch (error) {
+      this.logger?.error(`[checkQueueMessages] Error processing messages: ${error}`)
+    }
+  }
+
+  /**
+   * Get current active live mode message pickup session for a given connection
+   * @param connectionId
+   * @returns
+   */
+  private async getLocalliveSession(connectionId: string): Promise<MessagePickupSession | undefined> {
+    this.logger?.debug(`[getLocalliveSession] Verify current active live mode for connectionId ${connectionId}`)
+
+    try {
+      if (!this.agent) throw new Error('Agent is not defined')
+      return this.agent.messagePickup.getLiveModeSession({ connectionId })
+    } catch (error) {
+      this.logger?.error(`[getLocalliveSession] error in getLocalliveSession: ${error}`)
+    }
+  }
+
+  /**
+   * This method allow find record into DB to determine if the connectionID has a liveSession in another instance
+   * @param connectionId
+   * @returns liveSession object or false
+   */
+  private async getLiveSessionFromDB(connectionId: string): Promise<any | boolean> {
```

Comment:
```suggestion
  private async findLiveSessionInDb(connectionId: string): Promise<MessagePickupSession | undefined> {
```

Let's try to refactor this in order to return an object with the same prototype of MessagePickupSession, for consistency with the local one. Instead of returning true, it returns the object. If not found, instead of returning false it will return undefined.

Reply:
The function was refactored to return a MessagePickupSession object, and adjustments were made to the function that stores the sessions and the table where they are saved.